activerecord/lib/active_record/associations/association.rb

```diff
@@ -211,7 +211,7 @@ def foreign_key_present?
         # the kind of the class of the associated objects. Meant to be used as
         # a sanity check when you are about to assign an associated record.
         def raise_on_type_mismatch!(record)
-          unless record.is_a?(reflection.klass) || record.is_a?(reflection.class_name.constantize)
+          unless record.is_a?(reflection.klass) || (record.is_a?(reflection.class_name.constantize) rescue false)
```

Comment:
@dcrec1 you have to split up the one-liner. Using `safe_constantize` is preferred to `rescue false` as it doesn't hide programming errors:

```ruby
tpyo rescue false # => false
```

Something along the lines of:

```ruby
if klass = reflection.class_name.constantize
  record.is_a?(klass)
end
```